src/interfaces/wallet.cpp

```diff
@@ -254,13 +257,14 @@ WalletTxStatus MakeWalletTxStatus(const CWalletTx& wtx)
 
 WalletTxStatus MakeWalletTxStatus(AnonWallet* pAnonWallet, const uint256 &hash, const CTransactionRecord &rtx)
 {
+    LOCK(cs_main);
```

Comment:
This one doesn't have the caller locking it... however the caller is commented out.  So this routine might not ever be called.  Since this is an anonwallet version, it definitely raises the question as to why it's not used.  This might be a bigger problem then just the lack of the cs_main lock.

The question to investigate is "how does the different coin types function through the code calling these two routines"?